CRM/Core/Form/EntityFormTrait.php

```diff
@@ -202,12 +202,23 @@ protected function setTranslatedFields() {
   protected function addEntityFieldsToTemplate() {
     foreach ($this->getEntityFields() as $fieldSpec) {
       if (empty($fieldSpec['not-auto-addable'])) {
-        $element = $this->addField($fieldSpec['name'], [], CRM_Utils_Array::value('required', $fieldSpec));
+        $element = $this->addField($fieldSpec['name'], [], CRM_Utils_Array::value('required', $fieldSpec), FALSE);
```

Comment:
The FALSE here denotes 'dont' use legacy date field type' - we don't have date fields in other entityTrait forms hence this wasn't set earlier